src/main/java/domain/board/ChessBoard.java

```diff
@@ -0,0 +1,54 @@
+package domain.board;
+
+import domain.piece.Empty;
+import domain.piece.Piece;
+import domain.position.Position;
+import domain.position.Route;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ChessBoard {
+    private final Map<Position, Piece> board;
+
+    public ChessBoard(final Map<Position, Piece> board) {
+        this.board = new HashMap<>(board);
+    }
+
+    // todo: movePiece로 메서드명 변경
+    public void move(final Position source, final Position target) {
+        validateEmptyRoute(source, target);
+        validateLegalMove(source, target);
+        movePiece(source, target);
+    }
+
+    private void validateEmptyRoute(final Position source, final Position target) {
+        final Route route = Route.create(source, target);
+        final List<Position> positions = route.getRoute();
+        final boolean isPieceExist = positions.stream().anyMatch(board::containsKey);
+        if (isPieceExist) {
+            throw new IllegalArgumentException("중간에 말이 있어서 이동할 수 없습니다.");
+        }
```

Comment:
```suggestion
        final Route route = Route.create(source, target);
        if (route.blocked(board)) {
            throw new IllegalArgumentException("중간에 말이 있어서 이동할 수 없습니다.");
        }
```
꼭 위와 같이 할 필요는 없지만, 이런 식으로 route 이 로직을 대신 처리하도록 해보는건 어떨까요?

Reply:
많이 고민했던 부분이에요! 

'객체에게 메시지를 보내서 동작하게 한다'는 객체지향의 원칙에 따라 `Route`가 해당 책임을 부여하고, 물어보는 것이 좋다고 생각해요. 하지만 매개변수로 `board`를 전달한다면, 단순히 일급 컬렉션이던 `Route`가 꽤 많은 것을 알게 된다고 생각해요. 

미션을 설계하는 과정에서 `board`에 대한 다양한 설계를 고려했어요. `List<List<Position>>`, `List<Position>`, `Map<Position, Piece>` 등이 예시에요. `board` 설계가 변경되면 `Route`의 코드에 수정도 불가피해요. 

위 근거로 `ChessBoard`가 해당 책임을 갖도록 했어요. 하지만 코드가 상대적으로 지저분하기에 계속 고민하고 있었어요. 설계를 고정했고 앞으로 변경 가능성이 없다고 한다면 `Route`에게 해당 책임을 맡겨도 될 것 같아요. 

피케이의 의견이 궁금해요! 피케이라면 어떻게 하실 건가요?

Reply:
많은 고민을 해주셨네요!

저는 `Route`가 그정도는 알고 있어도 괜찮다고 생각해요. 현재는 `Route` 가 만들어지는 과정이 조금 복잡하긴 하지만, 생성 로직 외에는 일급컬렉션 자체에 로직이 따로 없어요. 일급 컬렉션을 만드는 이유 중 하나가 그 안에 있는 객체들을 활용한 로직을 한 곳에 모으는 것이기 때문에 `getRoute` 를 해서 처리하는 로직은 이 안에 넣는게 일급 컬렉션을 잘 활용하는 방식일 것 같네요.

board 의 설계 관련 부분은 그 형태가 다양할 수 있다는 것에 동의 해요. 그래서 아마 망쵸와 페어가 함께 가장 적절한 방식을 찾기 위해 논의하고 만들어주셨을거라 생각합니다. 그리고 제 생각에도 Map 을 사용한 방식이 좋다고 생각하고요. 그렇기 때문에 그 설계가 바뀔 가능성이 매우 낮다고 생각해요. Map 의 구현체가 바뀔 수는 있어도 Map 이 `List<List<Position>>` 으로 바뀔 가능성은 정말 낮다고 생각하는겁니다. `설계를 고정했고 앞으로 변경 가능성이 없다고 한다면 Route에게 해당 책임을 맡겨도 될 것 같아요.` 라는 말이 정확한 것 같네요. 👍 

그리고 바뀐다고 하더라도 어차피 `Route` 쪽에서는 이를 받아 stream 을 활용한 간단한 로직을 수행해요. 그래서 변경하기 그리 어려울 것 같진 않습니다!